pandas/core/nanops.py

```diff
@@ -812,15 +812,21 @@ def unique1d(values):
     return uniques
 
 
-def _checked_add_with_arr(arr, b):
+def _checked_add_with_arr(arr, b, arr_nans=None, b_nans=None):
```

Comment:
Not with good documentation they won't.  I'm not convinced by your consistency argument because you're going to sacrifice performance.  Every function in `nanops.py` calls `isnull`, but we have no reason to do this because we already know the `null` values before calling the function.  The other option is that we would call the specific attribute of the TDI object internally but then that makes no other universal to all `pandas` objects.